corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -141,42 +160,91 @@ hqDefine("geospatial/js/geospatial_map", [
                 currCase.marker = marker;
             };
 
-            function moveMarkerToClickedCoordinate(coordinates) {
-                if (clickedMarker != null) {
-                    clickedMarker.remove();
-                }
-                if (draw.getMode() === 'draw_polygon') {
-                    // It's weird moving the marker around with the ploygon
-                    return;
-                }
-                clickedMarker = new mapboxgl.Marker({color: "FF0000", draggable: true});
-                clickedMarker.setLngLat(coordinates);
-                clickedMarker.addTo(map);
-            }
-
             // Handle click events here
             map.on('click', (event) => {
                 let coordinates = getCoordinates(event);
             });
             return self;
         };
 
+        var exportGeoJson = function(drawInstance) {
+            // Credit to https://gist.github.com/danswick/36796153bd86ce982a59043cbe0ac8f7
+            var exportButton = $("#btnExport");
+            var data = drawInstance.getAll();
+
+            if (data.features.length) {
+                // Stringify the GeoJson
+                var convertedData = 'text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(data));
+
+                // Create export
+                exportButton.attr('href', 'data:' + convertedData);
+                exportButton.attr('download','data.geojson');
+            }
+        };
+
+        var mapControlsModel = function () {
+            'use strict';
+            var self = {};
+            var mapboxinstance = map.getMapboxInstance();
+            var polygons = [];
+            self.btnExportDisabled = ko.observable(true);
+
+            var mapHasPolygons = function() {
+                return polygons.length;
+            };
+
+            function removeId(id) {
+                var index = polygons.indexOf(id); // Find the index of the specified ID
+                polygons.splice(index, 1); // Remove the ID from the array
+              }
+
+            mapboxinstance.on('draw.delete', function(e) {
+                e.features.forEach((feature) => {
+                    if (feature.geometry.type == "Polygon") {
+                        removeId(feature.id);
```

Comment:
May be its possible to this without a function and a single line (like [this](https://stackoverflow.com/a/3596141)?)

Reply:
I refactored a bit (https://github.com/dimagi/commcare-hq/pull/33062/commits/82bd6ffc0571673436adbc91429b8f1e74b1f885), so this code isn't needed anymore